components/service/glean/scripts/sdk_generator.gradle

```diff
@@ -273,6 +275,56 @@ ext.gleanGenerateMetricsAPI = {
     }
 }
 
+// Generate the Metrics docs, if requested.
+if (project.ext.has("generateMarkdownDocs")) {
+    task gleanGenerateMetricsDocs(type: Exec) {
+        description = "Generate the Markdown docs for the collected metrics"
+
+        outputs.dir "$projectDir/docs"
+        workingDir rootDir
+
+        // Note that the command line is OS dependant: on linux/mac is Miniconda3/bin/python.
+        if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+            commandLine new File(GLEAN_MINICONDA_DIR, "python")
+        } else {
+            commandLine new File(GLEAN_MINICONDA_DIR, "bin/python")
+        }
+
+        args "-c"
+        args runPythonScript
+        args "glean_parser"
+        args GLEAN_PARSER_VERSION
+        args "translate"
+        args "-f"
+        args "markdown"
+        args "-o"
+        args "$projectDir/docs"
+        // We intentionally don't let out the local metrics.yaml, even if another one will
+        // be extracted from the AAR file. The parser will just ignore this one, if not present.
+        args "$projectDir/metrics.yaml"
+        args "$projectDir/pings.yaml"
+
+        // Only show the output if something went wrong.
+        ignoreExitValue = true
+        standardOutput = new ByteArrayOutputStream()
+        errorOutput = standardOutput
+        doLast {
+            println("$commandLine")
+            if (execResult.exitValue != 0) {
+                throw new GradleException("Process '${commandLine}' finished with non-zero exit value ${execResult.exitValue}:\n\n${standardOutput.toString()}")
+            }
+        }
+    }
+
+    // Attach the docs generation task to the code generation task, to make sure
+    // we generate docs, if we're asked for it.
+    tasks.whenTaskAdded { task ->
+        if (task.name.startsWith(gleanTaskNamePrefix)) {
+            task.dependsOn(gleanGenerateMetricsDocs)
+        }
+    }
+}
```

Comment:
This is a great point. I had the same idea and attempted to implement that first. Well, I failed :( The only workaround I found was to have `preBuild.finalizedBy(gleanGenerateMetricsDocs)`. However, I'm not sure `preBuild` is late enough. Unfortunately, I could not find a "postBuild". Thoughts?

Reply:
> 
> 
> `preBuild` might not be late enough because the Python environment may not yet exist? I guess you could always test that theory by deleting `~/.gradle/glean` and then rebuilding a clean checkout.

I just confirmed. Unfortunately `preBuild` is too early :'(

> Beyond that, I don't know if I have any great ideas. We could try something like just attaching it to the first variant rather than `android.applicationVariants.all`...

Unfortunately that wouldn't work as well :( The *first* variant isn't necessarily the currently active variant in the IDE :(

> But like I said, if finding a solution here is too cumbersome, I'm 💯 on just leaving this as-is.

I think I will do this :( sorry :(